app/src/main/java/ch/epfl/sweng/SDP/Constants.java

```diff
@@ -0,0 +1,31 @@
+package ch.epfl.sweng.SDP;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+public class Constants {
+    public FirebaseDatabase database;
+    public DatabaseReference databaseRef;
+    private DatabaseReference usersRef;
+    private FirebaseUser firebaseUser;
+
+    /**
+     * Defines Constants for Firebase.
+     */
+    public Constants() {
+        this.database = FirebaseDatabase.getInstance("https://gyrodraw.firebaseio.com/");
```

Comment:
You could use the new helper class from @Gondolav => `this.database = Database.getInstance()`

Reply:
Actually he can replace directly usersRef with `usersRef = Database.INSTANCE.getReference(« users »)`

Reply:
Your are right, thanks.

Reply:
Your are right, thanks.

Reply:
Your are right, thanks.

Reply:
Your are right, thanks.

Reply:
You are right, thanks.